lib/resources/windows_feature.rb

```diff
@@ -1,84 +1,126 @@
 # encoding: utf-8
 
-# check for a Windows feature
-# Usage:
-# describe windows_feature('DHCP Server') do
-#   it{ should be_installed }
-# end
-#
-# deprecated serverspec syntax:
-# describe windows_feature('IIS-Webserver') do
-#   it{ should be_installed.by("dism") }
-# end
-#
-# describe windows_feature('Web-Webserver') do
-#   it{ should be_installed.by("powershell") }
-# end
-#
-# This implementation uses the Get-WindowsFeature commandlet:
-# Get-WindowsFeature | Where-Object {$_.Name -eq 'XPS Viewer' -or $_.DisplayName -eq 'XPS Viewe
-# r'} | Select-Object -Property Name,DisplayName,Description,Installed,InstallState | ConvertTo-Json
-# {
-#     "Name":  "XPS-Viewer",
-#     "DisplayName":  "XPS Viewer",
-#     "Description":  "The XPS Viewer is used to read, set permissions for, and digitally sign XPS documents.",
-#     "Installed":  false,
-#     "InstallState":  0
-# }
 module Inspec::Resources
   class WindowsFeature < Inspec.resource(1)
     name 'windows_feature'
     supports platform: 'windows'
     desc 'Use the windows_feature InSpec audit resource to test features on Microsoft Windows.'
-    example "
-      describe windows_feature('dhcp') do
+    example <<-EOX
+      # By default this resource will use Get-WindowsFeature.
+      # Failing that, it will use DISM.
+
+      # Get-WindowsFeature Example
+      describe windows_feature('Web-WebServer', :powershell) do
+        it { should be_installed }
+      end
+
+      # DISM Example
+      describe windows_feature('IIS-WebServer', :dism) do
         it { should be_installed }
       end
-    "
 
-    def initialize(feature)
+      # Try PowerShell then DISM Example
+      describe windows_feature('IIS-WebServer') do
+        it { should be_installed }
+      end
+    EOX
+
+    def initialize(feature, method = nil)
       @feature = feature
+      @method = method
       @cache = nil
 
       # verify that this resource is only supported on Windows
       return skip_resource 'The `windows_feature` resource is not supported on your OS.' if !inspec.os.windows?
```

Comment:
I don't think we need this as we already have `supports platform: 'windows'` on line 6.

Reply:
Agreed! Bit out of scope but I'm always willing to clean up old code. Fixed in latest commit.